readthedocs/projects/migrations/0133_addons_standard_root_selector.py

```diff
@@ -0,0 +1,67 @@
+# Generated by Django 4.2.16 on 2024-11-14 13:16
+
+from django.db import migrations, models
+from django_safemigrate import Safe
+
+
+class Migration(migrations.Migration):
+    safe = Safe.before_deploy
+
+    dependencies = [
+        ('projects', '0132_addons_linkpreviews_fields'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name='addonsconfig',
+            name='doc_diff_root_selector',
+        ),
+        migrations.RemoveField(
+            model_name='addonsconfig',
+            name='linkpreviews_doctool_name',
+        ),
+        migrations.RemoveField(
+            model_name='addonsconfig',
+            name='linkpreviews_doctool_version',
+        ),
+        migrations.RemoveField(
+            model_name='addonsconfig',
+            name='linkpreviews_root_selector',
+        ),
+        migrations.RemoveField(
+            model_name='historicaladdonsconfig',
+            name='doc_diff_root_selector',
+        ),
+        migrations.RemoveField(
+            model_name='historicaladdonsconfig',
+            name='linkpreviews_doctool_name',
+        ),
+        migrations.RemoveField(
+            model_name='historicaladdonsconfig',
+            name='linkpreviews_doctool_version',
+        ),
+        migrations.RemoveField(
+            model_name='historicaladdonsconfig',
+            name='linkpreviews_root_selector',
+        ),
+        migrations.AddField(
+            model_name='addonsconfig',
+            name='options_doctool_name',
+            field=models.CharField(blank=True, choices=[('sphinx', 'Sphinx'), ('other', 'Other')], help_text='Name of the documentation tool used by this project', max_length=128, null=True),
+        ),
+        migrations.AddField(
+            model_name='addonsconfig',
+            name='options_doctool_root_selector',
+            field=models.CharField(blank=True, help_text='CSS selector for the main content of the page', max_length=128, null=True),
+        ),
+        migrations.AddField(
+            model_name='historicaladdonsconfig',
+            name='options_doctool_name',
+            field=models.CharField(blank=True, choices=[('sphinx', 'Sphinx'), ('other', 'Other')], help_text='Name of the documentation tool used by this project', max_length=128, null=True),
+        ),
+        migrations.AddField(
+            model_name='historicaladdonsconfig',
+            name='options_doctool_root_selector',
+            field=models.CharField(blank=True, help_text='CSS selector for the main content of the page', max_length=128, null=True),
+        ),
```

Comment:
Just noting that these are add/remove operations, and feel like they could be alters instead. It's probably not necessary, the docdiff selector is probably the only one potentially used by users so far. Maybe a migration is necessary for this at least?

Reply:
Good point. It seems we already have some users setting this value:

```
In [4]: list(AddonsConfig.objects.filter(doc_diff_root_selector__isnull=False, project__isnull=False).values_list('project__slug', 'doc_diff_root_selector'))
Out[4]: 
[('keptn', 'main.md-main'),
 ('unifier', '[role=main]'),
 ('okeamah', 'Master'),
 ('wpextract', '.md-main'),
 ('sphinx-docsearch', '[role=main]'),
 ('sphinxawesome-theme', '[role=main]'),
 ('cloudsigma-docs', '[role="master"]'),
 ('e-global-services', 'Main'),
 ('sphinx', '[role=main]')]
```